pkg/cfn/manager/delete_tasks.go

```diff
@@ -67,7 +66,11 @@ func (c *StackCollection) NewTasksToDeleteNodeGroups(shouldDelete func(string) b
 	if err != nil {
 		return nil, err
 	}
+	return c.NewTasksToDeleteNodeGroupsFromCFNDescriptions(shouldDelete, wait, cleanup, nodeGroupStacks)
+}
 
+// NewTasksToDeleteNodeGroupsFromCFNDescriptions defines tasks required to delete all of the nodegroups from a list of cloudformation stack descriptions
+func (c *StackCollection) NewTasksToDeleteNodeGroupsFromCFNDescriptions(shouldDelete func(string) bool, wait bool, cleanup func(chan error, string) error, nodeGroupStacks []*Stack) (*TaskTree, error) {
 	tasks := &TaskTree{Parallel: true}
 
 	for _, s := range nodeGroupStacks {
```

Comment:
> if !shouldDelete(name) {
			continue
		}

I think the filtering part should be moved into `NewTasksToDeleteNodeGroups()` so that the responsibility of this function is to just delete any task that is provided. That would be a better API IMO. What do you think?

Reply:
To clarify, do you mean moved out of rather than into? if anything passed to NewTasksToDeleteNodeGroups is to be deleted

I do think that would be better since its more succinct